pkg/integrations/v4/emitter/emitter_test.go

```diff
@@ -684,7 +684,121 @@ func Test_EmitV3_IntegrationNameVersion(t *testing.T) {
+		intDefinition,
+		extraLabels, // Is not in secure forward mode so Custom Attributes should not be sent
+		entityRewrite,
+		integration2.ProtocolV4.ParsedV4,
+	))
+
+	em := &VersionAwareEmitter{
+		aCtx:        mockAgent(false, customAttributes),
+		ffRetriever: feature_flags.NewManager(map[string]bool{fflag.FlagProtocolV4: true, fflag.FlagDMRegisterEnable: false}),
+		dmEmitter:   dmEmitter,
+	}
+
+	err := em.Emit(intDefinition, extraLabels, entityRewrite, integration2.ProtocolV4.Payload)
+	require.NoError(t, err)
+
+	dmEmitter.AssertExpectations(t)
+}
+
+func TestEmit_SendCustomAttributes_nilCustomAttributes(t *testing.T) {
+	intDefinition := integration.Definition{
+		InventorySource: *ids.NewPluginID("cat", "term"),
+	}
+	extraLabels := data.Map{
+		"label.foo":                "bar",
+		"extraAnnotationAttribute": "annotated",
+	}
+	entityRewrite := []data.EntityRewrite{}
+
+	dmEmitter := &mockDmEmitter{}
+	dmEmitter.On("Send", fwrequest.NewFwRequest(
+		intDefinition,
+		extraLabels,
+		entityRewrite,
+		integration2.ProtocolV4.ParsedV4,
+	))
+
+	em := &VersionAwareEmitter{
+		aCtx:        mockAgent(true, nil),
+		ffRetriever: feature_flags.NewManager(map[string]bool{fflag.FlagProtocolV4: true, fflag.FlagDMRegisterEnable: false}),
+		dmEmitter:   dmEmitter,
+	}
+
+	err := em.Emit(intDefinition, extraLabels, entityRewrite, integration2.ProtocolV4.Payload)
+	require.NoError(t, err)
+
+	dmEmitter.AssertExpectations(t)
+}
+
+func TestEmit_SendCustomAttributes_SendCAInSecureForwardMode(t *testing.T) {
+	intDefinition := integration.Definition{
+		InventorySource: *ids.NewPluginID("cat", "term"),
+	}
+	extraLabels := data.Map{
+		"label.foo":                "bar",
+		"extraAnnotationAttribute": "annotated",
+	}
+	customAttributes := config.CustomAttributeMap{
+		"customattributes.label.foo":               "bar",
+		"customAttributesExtraAnnotationAttribute": "annotated",
+	}
+	entityRewrite := []data.EntityRewrite{}
+
+	expectedLabels := make(data.Map)
+	for k, v := range extraLabels {
+		expectedLabels[k] = v
+	}
+	for k, v := range customAttributes {
+		expectedLabels[k] = v.(string)
+	}
+
+	dmEmitter := &mockDmEmitter{}
+	dmEmitter.On("Send", fwrequest.NewFwRequest(
+		intDefinition,
+		expectedLabels,
+		entityRewrite,
+		integration2.ProtocolV4.ParsedV4,
+	))
+
+	em := &VersionAwareEmitter{
+		aCtx:        mockAgent(true, customAttributes),
+		ffRetriever: feature_flags.NewManager(map[string]bool{fflag.FlagProtocolV4: true, fflag.FlagDMRegisterEnable: false}),
+		dmEmitter:   dmEmitter,
+	}
+
+	err := em.Emit(intDefinition, extraLabels, entityRewrite, integration2.ProtocolV4.Payload)
+	require.NoError(t, err)
+
+	dmEmitter.AssertExpectations(t)
+}
+
+func mockAgent2Payloads() *mocks.AgentContext {
+	ma := mockAgent(false, nil)
+	ma.On("SendData", mock.AnythingOfType("agent.PluginOutput")).Twice()
+	ma.SendDataWg.Add(1)
+
+	return ma
+}
+
+func mockAgent(isForwardOnly bool, customAttributes config.CustomAttributeMap) *mocks.AgentContext {
```

Comment:
Adding optional (`nil`-able) args makes the calls to this fn harder to read. Could we make `mocks.AgentContext` have a setter for `customAttributes` instead?

Reply:
[Options pattern](https://www.sohamkamani.com/golang/options-pattern/) would also be ideal, but perhaps an overkill for this case.

Reply:
Overkill for his case. I would revert the commit were I did that and duplicate the `mockAgent` code for the readiness sake of the rest of the tests.